### PR TITLE
Updates in flutter/assets for the shell refactor. (Patch 1)

### PR DESCRIPTION
**Pull request by chinmaygarde**
This changes is part of a large patch for easier review. Try the whole patch in one go by checking out https://github.com/chinmaygarde/flutter_engine/tree/shell directly. This patch contains the following changes:

* Created a single AssetResolver interface.
* Created an asset manager that implements this interface and manages multiple asset resolvers on behalf of an engine run configuration.

### CODE REVIEW

File: assets/file_asset_bundle.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+#ifndef FLUTTER_ASSETS_FILE_ASSET_BUNDLE_H_
+#define FLUTTER_ASSETS_FILE_ASSET_BUNDLE_H_
+
+#include "flutter/assets/asset_resolver.h"
+#include "lib/fxl/macros.h"
+
+namespace blink {
+
+class FileAssetBundle final : public AssetResolver {
```
Comment (jason-simmons) [inline assets/file_asset_bundle.h]: Is FileAssetBundle used anywhere?

Comment (chinmaygarde) [reply assets/file_asset_bundle.h]: Nope. Removed.

**Comment by cbracken** [comment]
lgtm

**Comment by chinmaygarde** [comment]
Addressed PR comments. PTAL.

**Comment by cbracken** [comment]
L even more GTM

**Comment by chinmaygarde** [comment]
Landed in https://github.com/flutter/engine/pull/4932.

**Pull request closed without merging**